tests/utils.py

```diff
@@ -0,0 +1,30 @@
+import time
+from typing import Any, Callable
+
+import snap_http
+from snap_http.types import SnapdResponse
+
+
+def call_and_await_api(name: str, *args: Any, **kwargs: Any) -> SnapdResponse:
+    """Call the `name` endpoint and wait until changes are applied."""
+    f: Callable[..., SnapdResponse] = getattr(snap_http, name)
+    response = f(*args, **kwargs)
+
+    if response.type == "sync":
+        return response
+
+    final_states = {"Done", "Abort", "Hold", "Error"}
+    change = response.change
+    while True:
+        time.sleep(1)
```

Comment:
Probably fine for now, but a minimum 1s sleep on every test might make things egregiously long eventually.

Reply:
Reduced to 0.1s